python/ray/tests/test_gcs_fault_tolerance.py

```diff
@@ -104,6 +104,56 @@ def test_gcs_server_restart_during_actor_creation(
     assert len(unready) == 0
 
 
+@pytest.mark.parametrize(
+    "ray_start_cluster_head_with_external_redis",
+    [
+        generate_system_config_map(
+            gcs_failover_worker_reconnect_timeout=2,
+            gcs_rpc_server_reconnect_timeout_s=60,
+            health_check_initial_delay_ms=0,
+            health_check_period_ms=1000,
+            health_check_failure_threshold=3,
+            enable_autoscaler_v2=True,
+        ),
+    ],
+    indirect=True,
+)
+def test_autoscaler_init(
+    ray_start_cluster_head_with_external_redis,
+):
+    """Checks that the node failure detector is correct when gcs server restart.
+
+    We set the cluster to timeout nodes after 2 seconds of heartbeats. We then
+    kill gcs server and remove the worker node and restart gcs server again to
+    check that the removed node will die finally.
+    """
+    cluster = ray_start_cluster_head_with_external_redis
+    worker = cluster.add_node()
+    cluster.wait_for_nodes()
+
+    # Make sure both head and worker node are alive.
+    nodes = ray.nodes()
+    assert len(nodes) == 2
+    assert nodes[0]["alive"] and nodes[1]["alive"]
+
+    head_node = cluster.head_node
+    gcs_server_process = head_node.all_processes["gcs_server"][0].process
+    gcs_server_pid = gcs_server_process.pid
+    # Kill gcs server.
+    cluster.head_node.kill_gcs_server()
+    # Wait to prevent the gcs server process becoming zombie.
+    gcs_server_process.wait()
+    wait_for_pid_to_exit(gcs_server_pid, 1000)
```

Comment:
this it 1 second? this might to too tight, and creates flaky tests on CI.

maybe change it to 10sec or something.

Reply:
nevermind, this seems to be 1000 sec..

Reply:
changed it to 300